### DISCUSSION TITLE
TabMenu is not updating its style when using unstyled mode and controlled mode

### DISCUSSION THREAD

**Discussion by sjunior96** [primefaces/community#1213] [PrimeReact] [answered]
Hi all,

I am using the TabMenu component from the PrimeReact + Tailwind and I noticed that when I use it in controlled mode and as unstyled, when the activeIndex change, it does not update the tab style, as shown below:

Before tab changes
![Captura de Tela 2024-02-17 às 00 48 35](https://github.com/primefaces/community/assets/36817476/62040ccc-90bd-4fef-aef3-585f9cf44253)

After tab changes
![Captura de Tela 2024-02-17 às 00 47 44](https://github.com/primefaces/community/assets/36817476/44f96791-3071-4597-808f-1eab713e1af4)

If we comment/remove the activeIndex and onTabChange props then it will work again.

Here is a Stackblitz example: https://stackblitz.com/edit/ze3hdg?file=src%2FApp.jsx

**Comment by melloware** [comment]
The bug is in the TailWind theme here:

`parent.state.activeIndex === context.index // Condition-based active styles.`

```js
tabmenu: {
        root: 'overflow-x-auto',
        menu: {
            className: classNames('flex m-0 p-0 list-none flex-nowrap', 'bg-white border-solid border-gray-300 border-b-2', 'outline-none no-underline text-base list-none')
        },
        menuitem: 'mr-0',
        action: ({ context, state }) => ({
            className: classNames(
                'cursor-pointer select-none flex items-center relative no-underline overflow-hidden',
                'border-b-2 p-5 font-bold rounded-t-lg ',
                'focus:outline-none focus:outline-offset-0 focus:shadow-[inset_0_0_0_0.2rem_rgba(191,219,254,1)] dark:focus:shadow-[inset_0_0_0_0.2rem_rgba(147,197,253,0.5)]',
                {
                    'border-gray-300 bg-white text-gray-700 hover:bg-white hover:border-gray-400 hover:text-gray-600 dark:bg-gray-900 dark:border-blue-900/40 dark:text-white/80 dark:hover:bg-gray-800/80': state.activeIndex !== context.index, // Condition-based hover styles.
                    'bg-white border-blue-500 text-blue-500 dark:bg-gray-900 dark:border-blue-300 dark:text-blue-300': state.activeIndex === context.index // Condition-based active styles.
                }
            ),
            style: { top: '2px' }
        }),
        icon: 'mr-2'
    },
```

I don't think in uncontrolled mode `parent.state.activeIndex` will always stay 0.

**Comment by melloware** [comment, marked as answer]
I opened a ticket from your discussion issue: https://github.com/primefaces/primereact/issues/5992

Fixed for 10.5.2+ !

**Reply by sjunior96** [reply]
That is great! We really appreciate it! We will wait anxious for this next release. God bless you!
